### PR TITLE
executor: join don't need any inner filter.

### PR DESCRIPTION
**Pull request by winoros**
You can see detail in [predicate_push_down.go](https://github.com/pingcap/tidb/blob/master/plan/predicate_push_down.go#L109).
Join won't retain the inner conditions.
PTAL @XuHuaiyu @zz-jason

### CODE REVIEW

**Comment by zz-jason** [comment]
should we also modify merge join ?

**Comment by winoros** [comment]
@zz-jason Merge join doesn't specify which side is the inner side.

File: executor/builder.go [hunk 1]
```diff
@@ -655,7 +654,6 @@ func (b *executorBuilder) buildHashJoin(v *plan.PhysicalHashJoin) Executor {
 	} else {
 		e.innerExec = rightExec
 		e.outerExec = leftExec
-		e.innerFilter = v.RightConditions
```
Comment (zz-jason) [inline executor/builder.go]: how about
```go
if len(v.RightConditions) != 0 {
    // throws an error like "hash join should not have any inner filter"
}
```

File: executor/join.go [hunk 2]
```diff
@@ -325,32 +324,21 @@ func (e *HashJoinExec) fetchOuterChunks(goCtx goctx.Context) {
 	}
 }
 
-// fetchSelectedInnerRows fetches all the selected rows from inner executor,
+// fetchInnerRows fetches all the selected rows from inner executor,
```
Comment (XuHuaiyu) [inline executor/join.go]: remove `selected`

File: executor/builder.go [hunk 3]
```diff
@@ -643,6 +643,9 @@ func (b *executorBuilder) buildHashJoin(v *plan.PhysicalHashJoin) Executor {
 	defaultValues := v.DefaultValues
 	lhsTypes, rhsTypes := leftExec.retTypes(), rightExec.retTypes()
 	if v.InnerChildIdx == 0 {
+		if len(v.LeftConditions) > 0 {
+			b.err = errors.Annotate(ErrBuildExecutor, "join's inner condition should be empty")
```
Comment (zz-jason) [inline executor/builder.go]: return nil

**Pull request merged**